### ISSUE TITLE
MPV/VLC Streams aren't playing in most cases

### ISSUE THREAD

**Issue by markus-k** [SoMuchForSubtlety/f1viewer#84]
I'm having trouble with f1viewer for playing sessions from F1TV with both MPV and VLC. I have a valid F1TV Pro subscription, playing in the web browser is no issue.

MPV reports HTTP error 403. VLC reports tons of ts demux errors. It's the same for both live and replay sessions.

In what feels like 1 out of 50 cases, the stream works, but most times it doesn't. Tried with latest release and current master, both last and this weekend.

**Comment by SoMuchForSubtlety** [comment]
Can you get the URL with the copy to clipboard option, then start MPV and VLC manually and post the log output here with something like pastebin.

**Comment by markus-k** [comment]
The log output from f1viewer only has the following (copying the currently running Q-Session Main Feed):
```
tail -n 3 ~/.local/share/f1viewer/2020-07-18.log
2020/07/18 15:07:14 [INFO] logged in!
2020/07/18 15:07:14 [INFO] checking for live session
2020/07/18 15:13:18 [INFO] URL copied to clipboard
```

MPV log:
```
$ mpv "https://f1tv-cdn-cent-live.formula1.com/live/wif0718043108/master.m3u8?nva=1595078118&ttl=86400&cc=DE&token=04eae4ecb908f51b035c5"
Playing: https://f1tv-cdn-cent-live.formula1.com/live/wif0718043108/master.m3u8?nva=1595078118&ttl=86400&cc=DE&token=04eae4ecb908f51b035c5
[ffmpeg] https: HTTP error 403 Forbidden
Failed to open https://f1tv-cdn-cent-west.formula1.com/live/wif0718043108/OTT_F1_WIF_AUDIO_0_64000.m3u8.
Can not open external file https://f1tv-cdn-cent-west.formula1.com/live/wif0718043108/OTT_F1_WIF_AUDIO_0_64000.m3u8.
[ffmpeg] https: HTTP error 403 Forbidden
Failed to open https://f1tv-cdn-cent-west.formula1.com/live/wif0718043108/OTT_F1_WIF_VIDEO_6_5500000.m3u8.


Exiting... (Errors when loading file)
```

VLC log:
```
$ vlc "https://f1tv-cdn-cent-live.formula1.com/live/wif0718043108/master.m3u8?nva=1595078330&ttl=86400&cc=DE&token=03ca18276eadca379716f"
VLC media player 3.0.8 Vetinari (revision 3.0.8-0-gf350b6b5a7)
[0000560764677630] main libvlc: Running vlc with the default interface. Use 'cvlc' to use vlc without interface.
[00007f4d58010bf0] ts demux error: libdvbpsi error (PSI decoder): TS duplicate (received 0, expected 1) for PID 0
[00007f4d58010bf0] ts demux error: libdvbpsi error (PSI decoder): TS duplicate (received 0, expected 1) for PID 480
[00007f4d58358f30] ts demux error: libdvbpsi error (PSI decoder): TS duplicate (received 0, expected 1) for PID 0
[00007f4d58358f30] ts demux error: libdvbpsi error (PSI decoder): TS duplicate (received 0, expected 1) for PID 480
[00007f4d58222980] ts demux error: libdvbpsi error (PSI decoder): TS duplicate (received 0, expected 1) for PID 0
[00007f4d58222980] ts demux error: libdvbpsi error (PSI decoder): TS duplicate (received 0, expected 1) for PID 480
[00007f4d58376660] ts demux error: libdvbpsi error (PSI decoder): TS duplicate (received 0, expected 1) for PID 0
[00007f4d58376660] ts demux error: libdvbpsi error (PSI decoder): TS duplicate (received 0, expected 1) for PID 480
[00007f4d584d0b80] ts demux error: libdvbpsi error (PSI decoder): TS duplicate (received 0, expected 1) for PID 0
(repeated a few hundred times...)
[00007f4d6c386bf0] main decoder error: buffer deadlock prevented
[00007f4d6c2cad50] mpeg4audio packetizer: AAC channels: 2 samplerate: 44100
[00007f4d74037430] ts demux error: libdvbpsi error (PSI decoder): TS duplicate (received 0, expected 1) for PID 0
[00007f4d74037430] ts demux error: libdvbpsi error (PSI decoder): TS duplicate (received 0, expected 1) for PID 480
....
```

This is on Ubuntu 18.04 from Germany.

**Comment by SoMuchForSubtlety** [comment]
Very strange. MPV can load the `master.m3u8` but it fails when loading the audio and video track. VLC seems to get further, but fails when loading the `.ts` segments.

My guess would be that cookies are not being set for the requests.

You can try to load the URL you get from f1viewer in a browser, open it in a text editor, load a track from that and open it in a text editor again, then you should see `.ts` segments.  The browser should set a cookie on the first request and use it on subsequent requests.

![image](https://user-images.githubusercontent.com/15961647/87854751-54f39580-c914-11ea-8f53-9b50d7fd46d8.png)

**Comment by markus-k** [comment]
That indeed seems to be the problem. Manually doing these requests in a browser or by supplying the set cookie to curl requests allows me to download valid `.ts` segments.

I also tried `mpv --cookies URL`, as the manpage says, cookies are disabled by disabled. But that doesn't help either.

After some digging I tried running mpv with `--msg-level=all=debug`, which shows it tries to use `youtube-dl` to grab the m3u8. When installing mpv I noticed it also installed `youtube-dl` from the Ubuntu repositories as a recommended dependency, so tried uninstalling it. And voilà: It works in mpv.

On to VLC: since f1viewer or the stream URL does not seem to be the problems now, something with VLC must be wrong. After a quick version check I uninstalled the Ubuntu version (3.0.8) and installed the current snap (3.0.11), and now even VLC works. Great!

The README should probably mention this then for Ubuntu 18.04 users: Don't install (an outdated?) youtube-dl from Ubuntu, and use the most up to date VLC.

Thank you! f1viewer itself seems to be at no fault whatsoever with this issue.
